content/300-guides/300-upgrade-guides/200-upgrading-versions/600-upgrading-to-prisma-5.mdx

```diff
@@ -0,0 +1,471 @@
+
+```prisma
+model Post {
+  id           String    @id @default(uuid())
+  commentsList Comment[]
+  tags         String[]
+}
+
+type Comment {
+  text String
+}
+```
+
+##### <inlinecode>equals</inlinecode> filter for composite lists
+
+[When using the `equals` filter with a composite list](https://www.prisma.io/docs/concepts/components/prisma-client/composite-types#finding-records-that-contain-composite-types-with-find-and-findmany) you will only be able to supply array values.
+
+Prisma 4 and lower:
+
+```js
+prisma.post.findMany({
+  where: {
+    commentsList: {
+      equals: { text: 'hello' },
+    },
+  },
+})
+```
+
+Prisma 5:
+
+```js highlight=4;normal
+prisma.post.findMany({
+  where: {
+    commentsList: {
+      equals: [{ text: 'hello' }],
+    },
+  },
+})
+```
+
+If you use the shorthand notation and exclude `equals`, you still must supply an array value for composite list fields.
+
+Prisma 4 and lower:
+
+```js
+prisma.post.create({
+  data: {
+    commentsList: { text: 'hello' },
+})
+
+prisma.post.findMany({
+  where: {
+    commentsList: { text: 'hello' },
+  },
+})
+```
+
+Prisma 5:
+
+```js highlight=3,8;normal
+prisma.post.create({
+  data: {
+    commentsList: [{ text: 'hello' }],
+})
+
+prisma.post.findMany({
+  where: {
+    commentsList: [{ text: 'hello' }],
+  },
+})
+```
+
+##### Scalar lists
+
+Scalar list values must be arrays in all operations.
+
+Prisma 4 and lower:
+
+```js
+prisma.post.create({
+  data: {
+    tags: 'databases',
+  },
+})
+
+prisma.post.findMany({
+  where: {
+    tags: 'databases',
+  },
+})
+```
+
+Prisma 5:
+
+```js highlight=3,9;normal
+prisma.post.create({
+  data: {
+    tags: ['databases'],
+  },
+})
+
+prisma.post.findMany({
+  where: {
+    tags: ['databases'],
+  },
+})
+```
+
+### Prisma Client embedded SQLite version updated
+
+With Prisma version 5.0.0, we have upgraded the embedded version of SQLite from `3.35.4` to `3.41.2`. We did not see any breaking changes and don't anticipate any changes needed in user projects, but if you are using SQLite, especially with raw queries that might go beyond Prisma's functionality, make sure to check [the SQLite changelog](https://www.sqlite.org/changes.html)
+
+### <inlinecode>jsonProtocol</inlinecode> out of preview!
+
+The `jsonProtocol` preview feature is now generally available. When upgrading to Prisma 5, make sure to remove `jsonProtocol` from your preview features, if added.
+
+Prisma 4 and lower:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["jsonProtocol", "tracing"]
+}
+```
+
+Prisma 5:
+
+```prisma highlight=3;delete
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["tracing"]
```

Comment:
Still not sure if we use the "highlight delete" in a way to signal that something was deleted from a line, or only if we want to show a line was removed and replaced with something else. I think usually we have a red think only if a link is completely dropped, and a green and red combination if we want to signal that something changed (without needing 2 clode blocks for it, like we have here).

Reply:
Yeah, I understand that. Trying to think through what this would look like. I'm going to use the previous example and just remove the line. I think in a small enough code block it will be obvious.